src/main/java/teammates/logic/api/EmailGenerator.java

```diff
@@ -399,6 +401,88 @@ private EmailWrapper generateFeedbackSessionEmailBaseForInstructorReminders(
                 EmailType.FEEDBACK_UNPUBLISHED.getSubject());
     }
 
+    public EmailWrapper generateFeedbackSessionResendLinksEmail(String userEmail) {
+        Instant endTime = new Date().toInstant();
+        Calendar calendar = Calendar.getInstance();
+        calendar.add(Calendar.MONTH, -6);
+        Instant startTime = calendar.getTime().toInstant();
+        String subject = EmailType.FEEDBACK_ACCESS_LINKS_RESENT.getSubject();
+        StringBuffer linksFragmentValue = new StringBuffer(1000);
+        String teammateHomePageLink = Config.APP_URL;
+        String emailBody;
+        String studentName = null;
+
+        List<FeedbackSessionAttributes> sessions = fsLogic.getAllOpenFeedbackSessions(startTime, endTime);
+        for (FeedbackSessionAttributes session : sessions) {
```

Comment:
Only open feedback sessions are returned by `getAllOpenFeedbackSessions` right? But the logic below is for closed and published sessions too?

Reply:
Actually the `getAllOpenFeedbackSessions` will also get those that are closed, as long as either `startTime` or `endTime` of a feedback session falls within the range. The code is like for the method: 
![image](https://user-images.githubusercontent.com/28507005/39344136-3ca34c5a-4a13-11e8-9a76-81ae2b1cb43e.png)
Also, I have tried to create a feedback session that is closed, and request access links, and part of the email is like this: 
![image](https://user-images.githubusercontent.com/28507005/39344165-655d515e-4a13-11e8-9368-597b2f629433.png)
which means the closed feedback session is also included.
Therefore, I think the method name for the database logic is misleading.

Reply:
I see. However, session published time is not taken into account right? E.g. a session that closed one year ago but was only published within the last 6 months.

Reply:
In that case, should I add a new method in `FeedbackSessionsDb` to get those sessions with publish date is within 6 months?

Reply:
Yup; I think you should write a whole new method for the new functionality and name it appropriately to avoid the confusion brought by `getAllOpenFeedbackSessions`.

Reply:
The new method should consist of two queries: one for start and end time (similar to the one in `getAllOpenFeedbackSessions`) and another one for publish time.